src/nvim/if_cscope.c

```diff
@@ -1420,12 +1436,12 @@ static char *cs_manage_matches(char **matches, char **contexts, int totmatches,
     break;
   case Free:
     if (mp != NULL) {
-      if (cnt > 0)
-        while (cnt--) {
-          xfree(mp[cnt]);
-          if (cp != NULL)
-            xfree(cp[cnt]);
-        }
+      while (cnt > 0) {
+        cnt--;
```

Comment:
I think `while (cnt--)` made more sense here. We don't need to worry about decrementing it below zero because we reset it after this loop.